internal/server/server_test.go

```diff
@@ -0,0 +1,71 @@
+package server_test
+
+import (
+	"testing"
+
+	"github.com/regmicmahesh/term-chat/internal/server"
+)
+
+func TestRegisterUser(t *testing.T) {
```

Comment:
I think it may be better to combine TestRegisterUser and TestIsUserRegistered, both seem similar to me, what do you say ?

Reply:
this tests needs to be done in another PR.